integration_tests/tests/billing-and-transaction-view-tests.ts

```diff
@@ -0,0 +1,188 @@
+    ${"BillingQty BillingPrice less than TransactionQty and TransactionPrice"}              | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME}        | ${"3"}             | ${"-3.3300"}  | ${"-1"} | ${"-33.3333"}          | ${"-33"}             | ${"6.6600"}  | ${"2"}     | ${"9.9900"}      | ${"3"}
+    ${"No TransactionQty No TransactionPrice(no events) but has BillingQty BillingPrice"}   | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.CURRENT_TIME}        | ${undefined}       | ${"6.6600"}   | ${"2"}  | ${undefined}           | ${undefined}         | ${"6.6600"}  | ${"2"}     | ${undefined}     | ${undefined}
+    ${"BillingQty less than TransactionQty and No BillingPrice but has TransactionPrice "}  | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client4"} | ${TimeStamps.CURRENT_TIME}        | ${11}              | ${"-27.5000"} | ${"-9"} | ${"-100.0000"}         | ${"-81"}             | ${"0.0000"}  | ${"2"}     | ${"27.5000"}     | ${"11"}
+    ${"BillingQty equals TransactionQty and No TransactionPrice No BillingPrice "}          | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client4"} | ${TimeStamps.CURRENT_TIME}        | ${2}               | ${"0.0000"}   | ${"0"}  | ${undefined}           | ${"0"}               | ${"0.0000"}  | ${"2"}     | ${"0.0000"}      | ${"2"}
+    ${"BillingQty greater than TransactionQty and No TransactionPrice but has BillingPrice"}| ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client4"} | ${TimeStamps.THIS_TIME_LAST_YEAR} | ${2}               | ${"27.0000"}  | ${"9"}  | ${undefined}           | ${"450"}             | ${"27.0000"} | ${"11"}    | ${"0.0000"}      | ${"2"}
+    ${"BillingQty equals TransactionQty but BillingPrice greater than TransactionPrice"}    | ${"IPV_PASSPORT_CRI_REQUEST_SENT"} | ${"client1"} | ${TimeStamps.THIS_TIME_LAST_YEAR} | ${2}               | ${"4.2000"}   | ${"0"}  | ${"170.7317"}          | ${"0"}               | ${"6.6600"}  | ${"2"}     | ${"2.4600"}      | ${"2"}
+    `(
+    "results retrived from billing and transaction_curated view query should match with expected $testCase,$billingQty,$priceDiff,$qtyDiff,$priceDifferencePercent,$qtyDifferencePercent,$billingPrice",
+    async ({
+      eventName,
+      clientId,
+      eventTime,
+      numberOfTestEvents,
+      priceDiff,
+      qtyDiff,
+      priceDifferencePercent,
+      qtyDifferencePercent,
+      billingPrice,
+      billingQty,
+      transactionPrice,
+      transactionQty,
+    }) => {
+      await assertResultsWithTestData(
+        eventName,
+        clientId,
+        eventTime,
+        numberOfTestEvents,
+        priceDiff,
+        qtyDiff,
+        priceDifferencePercent,
+        qtyDifferencePercent,
+        billingPrice,
+        billingQty,
+        transactionPrice,
+        transactionQty
+      );
```

Comment:
this is pretty hefty! 😅 

To tidy this up you could make `assertResultsWithTestData` take an argument called `data` (or something like that) where `data` is an object like 
```
{
      eventName,
      clientId,
      eventTime,
      numberOfTestEvents,
      priceDiff,
      qtyDiff,
      priceDifferencePercent,
      qtyDifferencePercent,
      billingPrice,
      billingQty,
      transactionPrice,
      transactionQty,
    }
```

That way your test predicate could just be 
```
async (data) => {
  await assertResultsWithTestData(data)
}
```

Reply:
do i need to change the test.each data structure to table  like                                                                                          
 const inputDataTable =[{testCase:"",}] then test.each(inputDataTable)("testdescription",async(data)